R/contrs.R

```diff
@@ -5,17 +5,29 @@
 #' @inheritParams stats::contr.sum
 #'
 #' @details
-#' In effects coding, unlike dummy coding ([stats::contr.treatment()]), each
-#' contrasts sums to 0. In regressions models, this results in an intercept that
+#' In effects coding, unlike treatment/dummy coding ([stats::contr.treatment()]), each
+#' contrast sums to 0. In regressions models, this results in an intercept that
 #' represents the (unweighted) average of the group means. In ANOVA settings,
 #' this also guarantees that lower order effects represent _main_ effects (and
-#' not _simple_ or _conditional_ effects, as is the case when using
-#' [stats::contr.treatment()], which is `R`'s default).
+#' not _simple_ or _conditional_ effects, as is the case when using R's default
+#' [stats::contr.treatment()]).
 #' \cr\cr
-#' `contr.deviation`, unlike [stats::contr.sum()], also has the added benefit
-#' that the factor-related coefficients are interpretable. In fact, they
-#' represent the same contrasts as those of [stats::contr.treatment()]: the
-#' difference of each level from the base level.
+#' Deviation coding (`contr.deviation`) is a type of effects coding.
+#' With deviation coding, the coefficients for factor variables are interpreted
+#' as the difference of each factor level from the base level
+#' (this is the same interpretation as with treatment/dummy coding). 
+#' For example, for a factor `group` with levels "Red" and "Blue", with `contr.devation`,
+#' the intercept represents the average of the group means for Red and Blue groups,
+#' and the coefficient for `groupBlue` represents the difference between Blue and Red
+#' group means.
+#' \cr\cr
+#' Sum coding ([stats::contr.sum()]) is another type of effects coding.
+#' With sum coding, the coefficients for factor variables are interpreted
+#' as the difference of each factor level from **the grand (across-groups) mean**.
+#' For example, for a factor `group` with levels "Red" and "Blue", with `contr.sum`,
+#' the intercept represents the average of the group means for Red and Blue groups,
+#' and the coefficient for `groupBlue` represents the difference of the Red group mean
+#' from the grand mean (the average of Red and Blue group means).
```

Comment:
I'm not sure I would say this. In a binary factor, the coefficient is half the distance of the second level from the grand mean, _and_ half the distance of the first level from the grand mean - or just the difference between the two levels.

```R
cbind(Intercept = 1, contr.sum(2)) |> 
  solve()
#>             1    2
#> Intercept 0.5  0.5
#>           0.5 -0.5
```

Reply:
Perhaps an example with 3 levels would be better?

Reply:
(I had a typo with Red and Blue there.) 

But I'm not sure why you are inverting the contrast matrix?

``` r
cbind(Intercept = 1, contr.sum(2))
#>   Intercept   
#> 1         1  1
#> 2         1 -1
```

The coefficient for each group with `contr.sum()` is the difference of that group's mean from the grand mean. With 2 levels, that would be half the value of the coefficient in treatment/deviation coding.

``` r
lm(mpg ~ factor(am), data = mtcars)
#> 
#> Call:
#> lm(formula = mpg ~ factor(am), data = mtcars)
#> 
#> Coefficients:
#> (Intercept)  factor(am)1  
#>      17.147        7.245

lm(mpg ~ factor(am), data = mtcars, contrasts = list(`factor(am)` = contr.sum))
#> 
#> Call:
#> lm(formula = mpg ~ factor(am), data = mtcars, contrasts = list(`factor(am)` = contr.sum))
#> 
#> Coefficients:
#> (Intercept)  factor(am)1  
#>      20.770       -3.622
```

An example with 3 levels would probably be clearer, yeah

Reply:
I'm inverting the matrix since it shows what the coefficients can be interpreted as - what difference (contrast) do they represent. To me, this makes it much more clear.